File: Data Structures (Java & C++)/Project 4/Project 4 Java/Node.java
/**
 * Project Name: Project 4 Java
 * Created by Alexandru Gutu on 11/1/2014.
 * Copyright (c) 2014 Alexandru Gutu. All rights reserved.
 */
public class Node
	{
		private int data;
		private Node left;
		private Node right;

		public Node(int data)
			{
				this.data = data;
//				this.counter = 1;
			}
		public int getData()
			{
				return this.data;
			}

		public Node getLeft()
			{
				return this.left;
			}

		public Node getRight()
			{
				return this.right;
			}

		public void setData(int data)
			{
				this.data = data;
			}

		public void setLeft(Node nextLeft)
			{
				this.left = nextLeft;
			}

		public void setRight(Node nextRight)
			{
				this.right = nextRight;
			}

		public void printData()
			{
				System.out.println(data);
			}

	}

File: Data Structures (Java & C++)/Project 4/Project 4 C++/Node.cpp
//
//  Node.cpp
//  Project 4 C++
//
//  Created by Alexandru Gutu on 11/14/14.
//  Copyright (c) 2014 Alexandru Gutu. All rights reserved.
//

#include "Node.h"

using namespace std;

Node::Node(int data)
{
	this->data = data;
	this->left = nullptr;
	this->right = nullptr;
}

const int Node::getData()
{
	return this->data;
}

Node *Node::getLeft()
{
	return this->left;
}

Node *Node::getRight()
{
	return this->right;
}

void Node::setData(int data)
{
	this->data = data;
}

void Node::setLeft(Node *nextLeft)
{
	this->left = nextLeft;
}

void Node::setRight(Node *nextRight)
{
	this->right = nextRight;
}

void Node::printData()
{
	cout << this->data << endl;
}